qiita_db/study.py

```diff
@@ -921,7 +957,52 @@ def ebi_submission_status(self, value):
 
     ebi_submission_status.__doc__.format(', '.join(_VALID_EBI_STATUS))
 
-    # --- methods ---
+    @property
+    def tags(self):
+        """Returns the tags of the study
+
+        Returns
+        -------
+        list of str
+            The study tags
+        """
+        with qdb.sql_connection.TRN:
+            sql = """SELECT study_tag_id, study_tag
+                        FROM qiita.study_tags
+                        LEFT JOIN qiita.per_study_tags USING (study_tag_id)
+                        WHERE study_id = {0}""".format(self._id)
+            qdb.sql_connection.TRN.add(sql)
+            return qdb.sql_connection.TRN.execute_fetchindex()
+
+    @tags.setter
+    def tags(self, tag_ids):
+        """Sets the tags of the study
+
+        Parameters
+        ----------
+        tag_ids : list of int
+            The tag ids of the study
+        """
+        with qdb.sql_connection.TRN:
+            sql = """DELETE FROM qiita.per_study_tags WHERE study_id = %s"""
+            qdb.sql_connection.TRN.add(sql, [self._id])
+            qdb.sql_connection.TRN.execute()
+
+            if tag_ids:
+                sql = """INSERT INTO qiita.per_study_tags
```

Comment:
If something goes wrong when this query is executed, we will have lost the tags 😱. IIRC the API should allow you to only execute at the very end of this function, such that if something fails the whole transaction is rolled back.